### PR TITLE
[refer #i1039] Replace log2(Double) with log2(BigInt)

### PR DESCRIPTION
**Pull request by knizhnik**

### CODE REVIEW

File: src/main/scala/org/ergoplatform/modifiers/history/PoPowAlgos.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ object PoPowAlgos {
       require(prevInterlinks.nonEmpty, "Interlinks vector could not be empty in case of non-genesis header")
       val genesis = prevInterlinks.head
       val tail = prevInterlinks.tail
-      val prevLevel = maxLevelOf(prevHeader)
+      val prevLevel = maxLevelOf(prevHeader) & 0xFF
```
Comment (gtklocker) [inline src/main/scala/org/ergoplatform/modifiers/history/PoPowAlgos.scala:39]: Why is this necessary?

Comment (abebeos) [reply src/main/scala/org/ergoplatform/modifiers/history/PoPowAlgos.scala:39]: @knizhnik , why no answer here?

**Comment by abebeos** [comment]
this PR affects a https://github.com/ergoplatform/ergo/labels/P0-critical issue.

Why is it unprocessed?

**Comment by abebeos** [comment]
@kushti , You can close this in favor of #1981

@knizhnik , you still need to explain the `0xFF` mask.
